src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -844,20 +854,35 @@
             // Execute stateful executable node analyzers, if any.
             if (executableNodeActions.Any())
             {
```

Comment:
>var syntaxNodesToAnalyzeBuilder = ArrayBuilder<SyntaxNode>.GetInstance(); [](http://example.com/codeflow?start=20&length=73)

I'll change this to specify the size. Note that one other negative thing about this copying data is if the array size exceeeds 128, this won't go back into the pool. Which makes this whole PR questionable if that is fairly common, as the intent here is to avoid allocations.

Reply:
Oops, this comment was intended for the other two ArrayBuilder.GetInstance calls in this method where the capacity is known beforehand.